### PR TITLE
fix: Set GNUPGHOME to a temp dir for project sync

### PR DESCRIPTION
**Pull request by bzwei**
AAP-24871: Project sync SCM clone failed Failed to import the gpg public key

### CODE REVIEW

**Review by Alex-Izquierdo — DISMISSED** [review]

File: src/aap_eda/services/project/scm.py [hunk 1]
```diff
@@ -211,6 +214,7 @@ def clone(
                 key_file.close()
             if gpg_key_file:
                 gpg_key_file.close()
+                gpg_home_dir.cleanup()
```
Comment (mkanoor) [inline src/aap_eda/services/project/scm.py:217]: @bzwei how about 
```
    if gpg_home_dir:
       gpg_home_dir.cleanup()
```

File: src/aap_eda/services/project/scm.py [hunk 2]
```diff
@@ -174,6 +175,8 @@ def clone(
             gpg_key_file.write("\n")
             gpg_key_file.flush()
             extra_vars["verify_commit"] = "true"
+            gpg_home_dir = tempfile.TemporaryDirectory()
+            env_vars["GNUPHOME"] = gpg_home_dir.name
```
Comment (hsong-rh) [inline src/aap_eda/services/project/scm.py:179]: it's GNUPGHOME, right?

File: src/aap_eda/services/project/scm.py (hunk 1 above)
Comment (bzwei) [reply src/aap_eda/services/project/scm.py:217]: gpg_key_file and gpg_home_dir are created in the same block. No need to add an if condition

**Comment by mkanoor** [comment]
Have we tested this given the first commit wasn't setting the correct parameter

**Comment by bzwei** [comment]
> Have we tested this given the first commit wasn't setting the correct parameter
Yes it was tested locally. The first commit had the wrong env, the key was neither imported to the temp dir, which produced a false positive test. Now both had been fixed and tested.

**Review by hsong-rh — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request merged**